fundraiser/project/fundraiser-contract/src/interface.sw

```diff
@@ -1,59 +1,16 @@
 library interface;
 
-dep data_structures;
+dep data_structures/asset_info;
+dep data_structures/campaign_info;
+dep data_structures/campaign;
+dep data_structures/pledge;
 
-use data_structures::{AssetInfo, Campaign, CampaignInfo, Pledge};
+use asset_info::AssetInfo;
+use campaign_info::CampaignInfo;
+use campaign::Campaign;
+use pledge::Pledge;
 
 abi Fundraiser {
-    /// Returns the number of unique assets that have been pledged across all campaigns
-    #[storage(read)]
-    fn asset_count() -> u64;
-
-    /// Returns information about the specificed asset, specifically if it has been added and the
-    /// pledged amount
-    ///
-    /// # Arguments
-    ///
-    /// * `asset` - Uniquie identifier that identifies the asset
-    #[storage(read)]
-    fn asset_info_by_id(asset: ContractId) -> AssetInfo;
-
-    /// Returns information about the specificed asset, specifically if it has been added and the
-    /// pledged amount
-    ///
-    /// The user interface will not know all possible assets that the contract contains therefore
-    /// this helper method allows the interface to iterate over the asset_count to discover all assets
-    ///
-    /// # Arguments
-    ///
-    /// * `index` - Number from 1...asset_count
-    #[storage(read)]
-    fn asset_info_by_count(index: u64) -> AssetInfo;
-
-    /// Returns information about the specified campaign for the campaign author
-    ///
-    /// # Arguments
-    ///
-    /// * `id` - Unique identifier which is a number starting from 1...storage.user_campaign_count
-    ///
-    /// # Reverts
-    ///
-    /// * When the `id` is either 0 or greater than the total number of campaigns created by the author
-    #[storage(read)]
-    fn campaign(campaign_history_index: u64, user: Identity) -> Campaign;
-
-    /// Returns information about the specified campaign
-    ///
-    /// # Arguments
-    ///
-    /// * `id` - Unique campaign identifier which is a number from the storage.total_campaigns range
-    ///
-    /// # Reverts
-    ///
-    /// * When the `id` is either 0 or greater than the total number of campaigns created
-    #[storage(read)]
-    fn campaign_info(id: u64) -> CampaignInfo;
-
     /// Marks a campaign as cancelled preventing further pledges or a claim to be made
```

Comment:
I'm not a huge fan of passing deadline (a block height) here.  Imo it seems annoying to a user (or frontend) to first have to query the block-height and then add it to a desired offset.  It seems better to take in an offset here and then add the block height internally in `CampaignInfo::new()`

Reply:
uhh `create_campaign` that is not `cancel_campaign` I must've put this comment in the wrong spot